src/main/java/net/neoforged/neoforge/common/extensions/IAttributeExtension.java

```diff
@@ -0,0 +1,217 @@
+     * The only vanilla attribute which performs value formatting is Knockback Resistance.<br>
+     *
+     * @param op    The operation of the modifier. Null if we are just displaying the raw value and not a modifier.
+     * @param value The value of the modifier.
+     * @param flag  The tooltip flag.
+     * @return The component form of the formatted value.
+     */
+    default MutableComponent toValueComponent(@Nullable Operation op, double value, TooltipFlag flag) {
+        // Knockback Resistance should be a PercentageAttribute, but we can't registry replace attributes, so we do this here.
+        // For Knockback Resistance, vanilla hardcodes a multiplier of 10 for addition values to hide numbers lower than 1,
+        // but representing the value as a percentage better expresses the intent of the modifier.
+        if (this == Attributes.KNOCKBACK_RESISTANCE.value()) {
+            return Component.translatable("neoforge.value.percent", FORMAT.format(value * 100));
+        }
+
+        // Speed has no metric, so displaying everything as percent works better for the user.
+        // However, Speed also operates in that the default is 0.1, not 1, so we have to special-case it instead of including it above.
+        if (this == Attributes.MOVEMENT_SPEED.value() && isNullOrAddition(op)) {
+            return Component.translatable("neoforge.value.percent", FORMAT.format(value * 1000));
+        }
+
+        // Default case for "normal" ranged attributes.
+        String key = isNullOrAddition(op) ? "neoforge.value.flat" : "neoforge.value.percent";
+        return Component.translatable(key, FORMAT.format(isNullOrAddition(op) ? value : value * 100));
+    }
+
+    /**
+     * Converts an attribute modifier into its tooltip representation.
+     * <p>
+     * This method does not handle formatting of "base" modifiers, such as Attack Damage or Attack Speed.
+     * <p>
+     * The returned component may append additional debug information based on the tooltip flag.
+     *
+     * @param modif The attribute modifier being converted to a component.
+     * @param flag  The tooltip flag.
+     * @return The component representation of the passed attribute modifier, with debug info appended if enabled.
+     */
+    default MutableComponent toComponent(AttributeModifier modif, TooltipFlag flag) {
+        Attribute attr = self();
+        double value = modif.amount();
+
+        MutableComponent comp;
+
+        if (value > 0.0D) {
+            comp = Component.translatable("neoforge.modifier.plus", this.toValueComponent(modif.operation(), value, flag), Component.translatable(attr.getDescriptionId())).withStyle(attr.getStyle(true));
+        } else {
+            value *= -1.0D;
+            comp = Component.translatable("neoforge.modifier.take", this.toValueComponent(modif.operation(), value, flag), Component.translatable(attr.getDescriptionId())).withStyle(attr.getStyle(false));
```

Comment:
Instead of duplicating this long line twice, can we just set a boolean variable for whether to pass `true` or `false` to `getStyle`?